src/traces/bar/set_positions.js

```diff
@@ -437,29 +436,12 @@ function updatePositionAxis(gd, pa, sieve, allowMinDtick) {
                 l = p + calcBarOffset,
                 r = l + calcBarWidth;
 
-            if(r >= maxPos) {
-                maxPos = r;
-                pMax = p;
-                pMaxOffset = calcBarOffset;
-                pMaxWidth = calcBarWidth;
-            }
-            if(l <= minPos) {
-                minPos = r;
-                pMin = p;
-                pMinOffset = calcBarOffset;
-                pMinWidth = calcBarWidth;
-            }
+            pMin = Math.min(pMin, l);
+            pMax = Math.max(pMax, r);
         }
     }
 
-    if(pMin) {
-        vpad = pMinWidth / 2;
-        Axes.expand(pa, [pMin + pMinOffset + vpad], {vpad: vpad});
-    }
-    if(pMax) {
-        vpad = pMaxWidth / 2;
-        Axes.expand(pa, [pMax + pMaxOffset + vpad], {vpad: vpad});
-    }
+    Axes.expand(pa, [pMin, pMax], {padded: false});
```

Comment:
@n-riesco would you mind sharing some info here about why this patch was needed? Thanks.